src/Directory.Build.props

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <!--
+        This file would be included in all projects automatically.
+        AssemblyCommon.props included only in SDK projects, because non-SDK projects does not support PackageReferences and some othrer properties.
+     -->
+    <Import Project="$(MSBuildThisFileDirectory)\Common\AssemblyCommon.props" Condition="'$(UsingMicrosoftNETSdk)' == 'true'" />
```

Comment:
Do we have non-.NET SDK project files that would break if AssemblyCommon.props was imported unconditionally?

Reply:
Non-SDK project is all of them that don't have `SDK` attribute in `Project` element. 
In this repository rough list of such projects are ([query](https://github.com/microsoft/qsharp-compiler/search?q=VSToolsPath)): `QSharpFileTemplate.csproj`, `QSharpAppTemplate.csproj`, `QSharpLibTemplate.csproj`, `QSharpTestTemplate.csproj`, `QSharpIonQAppTemplate.csproj`, `QSharpHoneywellAppTemplate.csproj`, `QSharpVsix.csproj`.

I think it's quite possible to change `AssemblyCommon.props` to work with those projects also with minor changes (like not including StyleCop reference by using condition). 
The reason I've done it this way is to keep behavior closer to what we had before since common properties were included only in SDK-based projects.

In the future, non-SDK project could be converted and this condition removed or `AssemblyCommon.props` could be updated to work with both. I think the first option is better because it sometimes hard to understand interactions between old and new properties (ex. `Platform` and `PlatfromTarget`) also old format quite verbose and hard to read.

Reply:
Ah, ok, right, the Visual Studio extension projects. That makes sense, thanks!